### PR TITLE
Detect and remove empty columns and rows while importing data

### PR DESCRIPTION
**Pull request by anastasia-mbithe**
Hello @rdstern  and @shadrackkibet I added the remove_empty() function  in the import dialog but it's still in progress. There are some issues to be sorted out before its ready for review. Thanks

### CODE REVIEW

**Comment by anastasia-mbithe** [comment]
@rdstern  and @shadrackkibet  you can have a look at it now, I feel its ready for reviews.

**Comment by lloyddewit** [comment]
@HawardKetoyoMsatsi Please could you peer review? thanks

**Review by HawardKetoyoMsatsi — CHANGES_REQUESTED** [review]

**Comment by anastasia-mbithe** [comment]
The functions were written as such before, am not sure if its fine changing all the function names in the dialog, so that it can be consistent with the other dialogs.

**Comment by lloyddewit** [comment]
> The functions were written as such before, am not sure if its fine changing all the function names in the dialog, so that it can be consistent with the other dialogs.

@HawardKetoyoMsatsi @anastasia-mbithe Thank you for the review and feedback. I agree that we should follow the naming conventions. For example, `clsDetectEmptyCols` is an RFunction so it should be `clsDetectEmptyColsFunction`. This increases readability. 

@anastasia-mbithe please could you ensure that all the newly added variables follow this convention? You are correct that there are many existing variables that don't follow this convention. We should change these if/when we refactor this dialog. In general, I don't think we should refactor dialogs whenever we make a small change to them. This would risk introducing regression.

@HawardKetoyoMsatsi Do you find the above a reasonable way ahead?
Thanks!

File: instat/dlgImportDataset.vb [hunk 1]
```diff
@@ -294,6 +306,8 @@ Public Class dlgImportDataset
         clsImportMultipleTextFiles = New RFunction
         clsFileNamesWithExt = New RFunction
```
[the remark this answers is not in the document]
Comment (lloyddewit) [reply instat/dlgImportDataset.vb:302]: I don't think we need to rename variables that already existed before this PR (see my comment below)

**Comment by anastasia-mbithe** [comment]
@lloyddewit and @HawardKetoyoMsatsi , I changed the naming of the two functions that were affected, you can have a look at it.

File: instat/dlgImportDataset.vb [hunk 2]
```diff
@@ -1100,5 +1126,26 @@ Public Class dlgImportDataset
...
+                clsTempFunction.bExcludeAssignedFunctionOutput = False
+                clsPipeOperator.AddParameter("y", clsRFunctionParameter:=clsTempFunction, iPosition:=0)
+                ucrBase.clsRsyntax.SetBaseROperator(clsPipeOperator)
+            Else
+                ucrBase.clsRsyntax.SetBaseRFunction(clsPreviousBaseFunction)
+            End If
+        Else
+            ucrChkDropEmptyCols.Visible = False
+            ucrBase.clsRsyntax.SetBaseRFunction(clsPreviousBaseFunction)
+        End If
```
Comment (lloyddewit) [inline instat/dlgImportDataset.vb:1145]: Is this function more readable if we invert the main 'If' (change the 'AndAlso's to 'OrElse's)?
Can we also add line breaks into the main 'If'?
```suggestion
        If strFileExtension = ".rds" _
                OrElse ucrBase.clsRsyntax.clsBaseFunction Is clsImportExcelMulti _
                OrElse ucrBase.clsRsyntax.clsBaseFunction Is clsImportMultipleFiles _
                OrElse ucrBase.clsRsyntax.clsBaseFunction Is clsImportMultipleTextFiles Then
            ucrChkDropEmptyCols.Visible = False
            ucrBase.clsRsyntax.SetBaseRFunction(clsPreviousBaseFunction)
        Else
            ucrChkDropEmptyCols.Visible = True
            If ucrChkDropEmptyCols.Checked Then
                Dim clsTempFunction As RFunction = clsPreviousBaseFunction.Clone
                clsTempFunction.RemoveAssignTo()
                clsTempFunction.bExcludeAssignedFunctionOutput = False
                clsPipeOperator.AddParameter("y", clsRFunctionParameter:=clsTempFunction, iPosition:=0)
                ucrBase.clsRsyntax.SetBaseROperator(clsPipeOperator)
            Else
                ucrBase.clsRsyntax.SetBaseRFunction(clsPreviousBaseFunction)
            End If
        End If
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]
@anastasia-mbithe this looks very good. I just have one comment

**Comment by anastasia-mbithe** [comment]
@lloyddewit  hello, I just made the change and still works fine. Thank you.

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern Please could you test? thanks

**Review by rdstern — APPROVED** [review]
I am happy to approve, though I note it only deletes variables with NA.  So empty character variables are still read.
I do not want issue 6883 to be closed though, because I still would like to be able to remove empty rows and variables (including empty character variables, once they are in R-Instat.  This could usefully be added to the Prepare > Delete Columns/Rows dialogue as suggested in this issue.

**Pull request merged**